### PR TITLE
feat: Basic plugin UI From

### PR DESCRIPTION
**Pull request by LiteSun**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**
Configuring the UI Form for popular plugins

- [x] api-breaker
- [x] limit-conn 
- [ ] cors
- [ ] fault-injection
- [x] basic-auth 
- [x] proxy-mirror
- [ ] limit-req
- [ ] limit-count
- [ ] traffic-split
- [x] referer-restriction

Please update this section with detailed description.

image

**Checklist:**

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
fine, due to this PR will get larger, @LiteSun Please resolve this one first. For other plugins, please send new PRs.

File: web/src/components/Plugin/UI/basic-auth.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import React, { forwardRef } from 'react';
+import type { FormInstance } from 'antd/es/form';
+import { Form, Input } from 'antd';
+
+type Props = {
+  form: FormInstance;
+  ref?: any;
+};
+
+const BasicAuth: React.FC<Props> = forwardRef(
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/basic-auth.tsx:26]: Why we need `forwardRef`?

File: web/src/components/Plugin/UI/plugin.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    case 'referer-restriction':
+      return <RefererRestriction form={form} />
+    case 'proxy-mirror':
+      return <ProxyMirror form={form} />
+    case 'api-breaker':
+      return <ApiBreaker form={form} />
+    default:
+      return <></>
+  }
+}
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/plugin.tsx:48]: note here

File: web/src/components/Plugin/UI/plugin.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    case 'basic-auth':
+      return <BasicAuth form={form} />
+    case 'referer-restriction':
+      return <RefererRestriction form={form} />
+    case 'proxy-mirror':
+      return <ProxyMirror form={form} />
+    case 'api-breaker':
+      return <ApiBreaker form={form} />
+    default:
+      return <></>
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/plugin.tsx:46]:
```suggestion
      return null
```

File: web/src/components/Plugin/UI/plugin.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { default as ApiBreaker } from './api-breaker'
+
+type Props = {
+  name: string,
+  form: FormInstance
+}
+
+export const PLUGIN_UI_LIST = ['basic-auth', 'limit-conn', 'referer-restriction', 'proxy-mirror', 'api-breaker'];
+
+export const PLUGIN_UI_FORM: React.FC<Props> = ({ name, form }) => {
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/plugin.tsx:33]:
```suggestion
export const PluginForm: React.FC<Props> = ({ name, form }) => {
```

File: web/src/components/Plugin/UI/plugin.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import { default as RefererRestriction } from './referer-restriction'
+import { default as ProxyMirror } from './proxy-mirror'
+import { default as ApiBreaker } from './api-breaker'
+
+type Props = {
+  name: string,
+  form: FormInstance
+}
+
+export const PLUGIN_UI_LIST = ['basic-auth', 'limit-conn', 'referer-restriction', 'proxy-mirror', 'api-breaker'];
```
Comment (juzhiyuan) [inline web/src/components/Plugin/UI/plugin.tsx:31]:
```suggestion
export const PLUGIN_LIST = ['basic-auth', 'limit-conn', 'referer-restriction', 'proxy-mirror', 'api-breaker'];
```

**Comment by LiteSun** [comment]
> fine, due to this PR will get larger, @LiteSun Please resolve this one first. For other plugins, please send new PRs.

OK

**Comment by moonming** [comment]
One PR only does one feature

**Comment by LiteSun** [comment]
> One PR only does one feature

ok

File: web/src/components/Plugin/UI/plugin.tsx (hunk 5 above)
Comment (LiteSun) [reply web/src/components/Plugin/UI/plugin.tsx:31]: `PLUGIN_UI_LIST ` variable names are more intuitive imo 🤔

**Comment by LiteSun** [comment]
This pr is too big and difficult to review, I'll break it down into smaller pr. Thanks for the reviewers.

**Pull request closed without merging**